src/builder.rs

```diff
@@ -193,11 +156,88 @@ impl<S> AstarteOptions<S> {
     }
 
     /// Add all the interfaces from the `.json` files contained in the specified folder.
-    pub fn interface_directory(self, interfaces_directory: &str) -> Result<Self, OptionsError> {
+    pub fn interface_directory(self, interfaces_directory: &str) -> Result<Self, BuilderError> {
         walk_dir_json(interfaces_directory)?
             .iter()
             .try_fold(self, |acc, path| acc.interface_file(path))
     }
+
+    /// This method configures the bounded channel size.
+    ///
+    pub fn channel_size(mut self, size: usize) -> Self {
+        self.channel_size = size;
+
+        self
+    }
+
+    /// Set the backing storage for the device.
+    ///
+    /// This will store and retrieve the device's properties.
+    pub fn store<T>(self, store: T) -> DeviceBuilder<T, C>
+    where
+        T: PropertyStore,
+    {
+        DeviceBuilder {
+            channel_size: self.channel_size,
+            interfaces: self.interfaces,
+            connection: self.connection,
+            store: StoreWrapper::new(store),
+        }
+    }
+
+    /// Establishes the connection using the passed [`ConnectionConfig`].
+    ///
+    /// If the connection gets established correctly, the caller can than construct
+    /// the [`crate::AstarteDeviceSdk`] using the [`DeviceSdkBuild::build`] method.
+    pub async fn connect<T>(self, config: T) -> Result<DeviceBuilder<S, T::Con>, crate::Error>
+    where
+        S: PropertyStore,
+        T: ConnectionConfig,
+        C: Send + Sync,
+    {
+        let connection = config.connect(&self).await?;
+
+        Ok(DeviceBuilder {
+            channel_size: self.channel_size,
+            interfaces: self.interfaces,
+            connection,
+            store: self.store,
+        })
+    }
+}
+
+impl<S, C> Debug for DeviceBuilder<S, C> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("AstarteOptions")
+            .field("interfaces", &self.interfaces)
+            // We manually implement Debug for the store, so we can avoid have a trait bound on
+            // `S` to implement [Display].
+            .finish_non_exhaustive()
+    }
+}
+
+impl Default for DeviceBuilder<(), ()> {
```

Comment:
`Default` it's a way to initialize the memory of the struct with a valid value. Having an impl for this trait allows the user of the sdk to use `std::mem::take` and related functions so I guess it could be useful.